diff --git a/test/legacy_test/test_api_compatibility_part2.py b/test/legacy_test/test_api_compatibility_part2.py
index 38d6368318..7a3a28d24d 100644
--- a/test/legacy_test/test_api_compatibility_part2.py
+++ b/test/legacy_test/test_api_compatibility_part2.py
@@ -1593,7 +1593,7 @@ class TestTrapezoidAPI(unittest.TestCase):
         paddle.trapezoid(y, out=out4)
 
         # Verify outputs
-        ref_out = np.trapezoid(self.np_y)
+        ref_out = np.trapz(self.np_y)
         for out in [out1, out2, out3, out4]:
             np.testing.assert_allclose(out.numpy(), ref_out, rtol=1e-5)
 
@@ -1620,7 +1620,7 @@ class TestTrapezoidAPI(unittest.TestCase):
                 fetch_list=[out1, out2, out3],
             )
 
-            ref_out = np.trapezoid(self.np_y)
+            ref_out = np.trapz(self.np_y)
             for actual in fetches:
                 np.testing.assert_allclose(actual, ref_out, rtol=1e-5)
 
diff --git a/test/legacy_test/test_randint_op.py b/test/legacy_test/test_randint_op.py
index 7fb866b75d..7bd6b809f2 100644
--- a/test/legacy_test/test_randint_op.py
+++ b/test/legacy_test/test_randint_op.py
@@ -171,7 +171,7 @@ class TestRandomValue(unittest.TestCase):
             return
 
         # Different GPU generatte different random value. Only test V100 here.
-        if "V100" not in paddle.device.get_device_name():
+        if "V100" not in paddle.device.get_device_name() or is_custom_device():
             return
 
         print("Test Fixed Random number on GPU------>")
